File: 125-valid-palindrome/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
const isPalindrome = function(s) {
    const str = s.replace(/[^a-z0-9]/gi, '').toLowerCase();
    for (let l = 0, r = str.length - 1; l < r; l++, r--) {
        if (str[l] !== str[r]) return false;
    }
    return true;
}

File: easy/125-valid-palindrome/125-valid-palindrome.rb
# @param {String} s
# @return {Boolean}
def is_palindrome(s)
    alphanumeric = s.gsub(/([^\w]|_)/, '').downcase
    p alphanumeric
    l = 0
    r = alphanumeric.length - 1
    while l < r do 
        return false if alphanumeric[l] != alphanumeric[r]
        l += 1
        r -= 1
    end
    true
end